raster/src/main/scala/geotrellis/raster/io/geotiff/writer/TiffTagFieldValue.scala

```diff
@@ -77,6 +79,30 @@ object TiffTagFieldValue {
     fieldValues += TiffTagFieldValue(PlanarConfigurationTag, ShortsFieldType, 1, PlanarConfigurations.PixelInterleave)
     fieldValues += TiffTagFieldValue(SampleFormatTag, ShortsFieldType, 1, imageData.bandType.sampleFormat)
 
+    if(geoTiff.options.colorSpace == ColorSpace.Palette) {
+
+      val bitsPerSample = imageData.bandType.bitsPerSample
+      val divider = 1 << bitsPerSample
+
+      for {
+        cmap ← geoTiff.options.colorMap
+        palette = IndexedColorMap.toTiffPalette(cmap)
+        size = math.min(palette.size, divider)
```

Comment:
Using `=` inside a `for` comprehension isn't idiomatic to the library. Might cause confusion because if it's in the for it seems like it should be a `map` or `flatMap`, that's how I read it anyway. Could be extracted to `val`s inside the body.